### PR TITLE
Add table function format(format_name, data)

### LINKED ISSUE #33812: Table function(s) to parse formatted data from string literal.
**Proposal**

```
SELECT * FROM data(JSONEachRow, 
$$
{"a": "Hello", "b": 111}
{"a": "World", "b": 123}
$$)
```

or

```
SELECT * FROM format(JSONEachRow, 
$$
{"a": "Hello", "b": 111}
{"a": "World", "b": 123}
$$)
```

or

```
SELECT * FROM JSONEachRow(
$$
{"a": "Hello", "b": 111}
{"a": "World", "b": 123}
$$)
```

did not decide yet.

### PR DESCRIPTION
**Pull request by Avogar**
Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add table function `format(format_name, data)`.

Detailed description / Documentation draft:
Table function file `format(format_name, data)`, where `data` argument is a string in `format_name` format. This table function It extracts table structure from `data` and parse it according to `format_name`.
Examples:
```sql
:) select * from format(JSONEachRow, 
$$
{"a": "Hello", "b": 111}
{"a": "World", "b": 123}
{"a": "Hello", "b": 112}
{"a": "World", "b": 124}
$$)

SELECT *
FROM format(JSONEachRow, '\n{"a": "Hello", "b": 111}\n{"a": "World", "b": 123}\n{"a": "Hello", "b": 112}\n{"a": "World", "b": 124}\n')

Query id: e009c5f8-f4f4-44bb-a6d0-c2638d9bf42d

┌───b─┬─a─────┐
│ 111 │ Hello │
│ 123 │ World │
│ 112 │ Hello │
│ 124 │ World │
└─────┴───────┘

4 rows in set. Elapsed: 0.015 sec.
```
```sql
:) desc format(JSONEachRow, 
$$
{"a": "Hello", "b": 111}
{"a": "World", "b": 123}
{"a": "Hello", "b": 112}
{"a": "World", "b": 124}
$$)

DESCRIBE TABLE format(JSONEachRow, '\n{"a": "Hello", "b": 111}\n{"a": "World", "b": 123}\n{"a": "Hello", "b": 112}\n{"a": "World", "b": 124}\n')

Query id: a92a24ff-4038-4591-aa25-c93cd552130c

┌─name─┬─type──────────────┬─default_type─┬─default_expression─┬─comment─┬─codec_expression─┬─ttl_expression─┐
│ b    │ Nullable(Float64) │              │                    │         │                  │                │
│ a    │ Nullable(String)  │              │                    │         │                  │                │
└──────┴───────────────────┴──────────────┴────────────────────┴─────────┴──────────────────┴────────────────┘

2 rows in set. Elapsed: 0.008 sec.
```
Closes https://github.com/ClickHouse/ClickHouse/issues/33812

### CODE REVIEW

File: src/TableFunctions/TableFunctionFormat.cpp [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    {
+        for (size_t i = 0; i != result.columns(); ++i)
+        {
+            auto & result_column = result.getByPosition(i);
+            const auto & new_column = new_block.getByPosition(i);
+            auto mutable_column = IColumn::mutate(result_column.column);
+            mutable_column->insertManyFrom(*new_column.column, 0, new_column.column->size());
+            result_column.column = std::move(mutable_column);
+        }
+    }
```
Comment (vdimir) [inline src/TableFunctions/TableFunctionFormat.cpp:88]: Is it the same as here?
I wonder why we don't have function for concatenating blocks, but actually it's only 5-10 LOC, so it's ok to duplicate it.
https://github.com/ClickHouse/ClickHouse/blob/02eb2fae1fd8d3c79fe79047a2c5973d1f8815a5/src/Interpreters/SortedBlocksWriter.cpp#L325-L340

File: tests/queries/0_stateless/02188_table_function_file.sql [hunk 2]
```diff
@@ -0,0 +1,68 @@
+select * from format(JSONEachRow,
```
Comment (vdimir) [inline tests/queries/0_stateless/02188_table_function_file.sql:1]: What's for `02188_table_function_file.sql` ? Is it should be only `02188_table_function_format.sql` ?

**Review by vdimir — APPROVED** [review]
LGTM

File: src/TableFunctions/TableFunctionFormat.cpp (hunk 1 above)
Comment (Avogar) [reply src/TableFunctions/TableFunctionFormat.cpp:88]: I will add a function for blocks concatination

File: tests/queries/0_stateless/02188_table_function_file.sql (hunk 2 above)
Comment (Avogar) [reply tests/queries/0_stateless/02188_table_function_file.sql:1]: It should be only `02188_table_function_format.sql`, just forgot to delete old file after bad renaming

**Pull request merged**